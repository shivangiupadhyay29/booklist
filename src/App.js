import React from "react";
import SearchBox from "./components/search-box";
import List from "./components/list";
import "./App.css";

function App() {
  return (
    <div className="container">
      <SearchBox />
      <List />
    </div>
  );
}

export default App;
