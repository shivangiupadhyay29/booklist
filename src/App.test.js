import React from 'react';
import App from './App';
import SearchBox from './components/search-box';
import BookList from './components/book-list';
import { shallow, mount } from 'enzyme';
import { render } from '@testing-library/react';

describe('root app component', () => {
  test('renders root component with child components', () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toBeTruthy();
    expect(wrapper.length).toBe(1);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(SearchBox)).toHaveLength(1);
    expect(wrapper.find(BookList)).toHaveLength(1);
  });

  test('sets the initial book list with empty array', () => {
    const { container } = render(<App />);
    expect(container).toBeTruthy();
  });

  test('mounting the component', () => {
    let wrapper;
    beforeEach(() => {
      wrapper = mount(<App />);
    });
    expect(wrapper).not.toBeNull();
  });
});
