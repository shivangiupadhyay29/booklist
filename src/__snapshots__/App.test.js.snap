// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`root app component renders root component with child components 1`] = `ShallowWrapper {}`;
